src/redux/BidData/index.ts

```diff
@@ -98,6 +137,18 @@ export const fetchSaleBids = (saleId: string, saleType: saleType, saleBidsReques
   }
 }
 
+export const fetchBidsFromChain = (bids: SaleBid): any => {
+  return async (dispatch: any) => {
+    dispatch(updateBidRequest(true))
+    try {
+      dispatch(updateBidSuccess(bids))
+    } catch (error) {
+      console.log(error)
+      dispatch(updateBidFailure(error))
+    }
+  }
+}
+
 const keyFinder = (object: BidsBySaleId) => {
   return Object.keys(object)[0]
 }
```

Comment:
`Object.keys` returns an array of the keys, fair. However, this assumes that the object key order is like 
```typescript 
{
  id: '0x',
  bids: [],
  createdAt: 1620750568
  updatedAt: 1620750568
}
```

[In JavaScript, the order of keys is not guaranteed](https://stackoverflow.com/a/5525820/2151050). Review this implementation.

Reply:
The data is transformed and received as an `action.payload` as this
```javascript
{
0x9b564bf6754acb06a6006eb0143e8235bc233d72:
    bids: (100) [{…}, ...]
    lastUpdated: 1620778192402
}
```

If there will always only be a single object that contains all the bids for that sale in the action payload then I do not need to worry about the order of the keys as there is only a single key.

why are we using es5?. Can't we switch to more recent versions like es2020, if we do that than I can just use `Object.getOwnPropertyNames ` as that does obey property creation order for keys.

Any other ideas would be helpful.